src/assets/Generator.Shared/ArmOperationHelpers.cs

```diff
@@ -0,0 +1,389 @@
+                this.DefaultWaitForCompletionAsync(OperationHelpers.DefaultPollingInterval, cancellationToken);
+
+            public override ValueTask<Response<T>> WaitForCompletionAsync(TimeSpan pollingInterval, CancellationToken cancellationToken) =>
+                this.DefaultWaitForCompletionAsync(pollingInterval, cancellationToken);
+
+            public override async ValueTask<Response> UpdateStatusAsync(CancellationToken cancellationToken = default)
+            {
+                if (HasCompleted)
+                {
+                    return GetRawResponse();
+                }
+
+                if (_shouldPoll)
+                {
+                    _rawResponse = await GetResponseAsync(_info.PollUri, cancellationToken).ConfigureAwait(false);
+                }
+                _shouldPoll = true;
+                _hasCompleted = IsTerminalState(out string state);
+                if (HasCompleted)
+                {
+                    Response finalResponse = GetRawResponse();
+                    if (s_failureStates.Contains(state))
+                    {
+                        throw await _clientDiagnostics.CreateRequestFailedExceptionAsync(finalResponse).ConfigureAwait(false);
+                    }
+
+                    if (_info.FinalUri != null)
+                    {
+                        finalResponse = await GetResponseAsync(_info.FinalUri, cancellationToken).ConfigureAwait(false);
+                    }
+                    switch (finalResponse.Status)
+                    {
+                        case 200:
+                        case 204 when !(_info.RequestMethod == RequestMethod.Put || _info.RequestMethod == RequestMethod.Patch):
+                        {
+                            Response<T> typedResponse =  await _createFinalResponseAsync(finalResponse, cancellationToken).ConfigureAwait(false);
+                            _rawResponse = typedResponse.GetRawResponse();
+                            _value = typedResponse.Value;
+                            _hasValue = true;
+                            break;
+                        }
+                        default:
+                            throw await _clientDiagnostics.CreateRequestFailedExceptionAsync(finalResponse).ConfigureAwait(false);
+                    }
+                }
+
+                return GetRawResponse();
+            }
+
+            public override Response UpdateStatus(CancellationToken cancellationToken = default)
+            {
+                if (HasCompleted)
+                {
+                    return GetRawResponse();
+                }
+
+                if (_shouldPoll)
+                {
+                    _rawResponse = GetResponse(_info.PollUri, cancellationToken);
+                }
+                _shouldPoll = true;
+                _hasCompleted = IsTerminalState(out string state);
+                if (HasCompleted)
```

Comment:
Don't use property and field interchangeably in the same method.

Reply:
Done.